test/topology_custom/test_mv_loss_attrs.py

```diff
@@ -0,0 +1,97 @@
+#
+from test.pylib.manager_client import ManagerClient
+
+import pytest
+import time
+import logging
+import glob
+import os
+
+from test.topology.util import reconnect_driver
+from test.pylib.util import wait_for_view
+
+logger = logging.getLogger(__name__)
+
+@pytest.mark.asyncio
+async def test_mv_loss_attrs_exception(manager: ManagerClient) -> None:
+    node_count = 2
+    servers = await manager.servers_add(node_count)
+    cql, _ = await manager.get_ready_cql(servers)
+    await cql.run_async("CREATE KEYSPACE ks WITH replication = {'class': 'NetworkTopologyStrategy', 'replication_factor': 2} AND tablets = {'enabled': false};")
+    # Create a table with non-primary key columns as collections
+    await cql.run_async("""CREATE TABLE ks.tab (base_key int, view_key int, ":attrs" map<text, text>, PRIMARY KEY (base_key, view_key)) WITH gc_grace_seconds = 1 AND tombstone_gc = {'mode': 'timeout'};""")
+    # Create a MV
+    await cql.run_async("CREATE MATERIALIZED VIEW ks.mv_cf_view AS SELECT * FROM ks.tab WHERE view_key IS NOT NULL and base_key IS NOT NULL PRIMARY KEY (view_key, base_key);")
+
+    await wait_for_view(cql, 'mv_cf_view', node_count)
+
+    # Simulate Alternator to perform the operation of deleting first and then writing
+    await cql.run_async("DELETE FROM ks.tab WHERE base_key = 5 AND view_key = 5;")
+    await cql.run_async('INSERT INTO ks.tab (base_key, view_key, ":attrs") VALUES (5, 5, {\'attr1\': \'aaaaa\'});')
+
+    # waite for gc_grace_seconds
+    time.sleep(10)
+
+    for server in servers:
+        await manager.api.keyspace_flush(server.ip_addr, 'ks')
+
+    # when node A compacted, the row inlcue liveness_info and deletion_info
+    # Node A is compacted first, but node B has not yet been compacted.
+    await manager.api.keyspace_compaction(servers[0].ip_addr, 'ks', 'tab')
+
+    # query the MV will be found the item
+    for row in cql.execute(f"SELECT * FROM ks.mv_cf_view WHERE base_key = 5 AND view_key = 5;"):
+        assert row.base_key == 5
+        assert row.view_key == 5
+        assert row.attrs['attr1'] == 'aaaaa'
+
+    def delete_files_with_prefix(directory, prefix='me-'):
+        for filename in os.listdir(directory):
+            if filename.startswith(prefix):
+                file_path = os.path.join(directory, filename)
+                try:
+                    if os.path.isfile(file_path):
+                        os.remove(file_path)
+                        print(f"Deleted: {file_path}")
+                except Exception as e:
+                    print(f"Error deleting {file_path}: {e}")
+
+    # Simulate MV asynchronous write failure
+    for server in servers:
+        node_workdir = await manager.server_get_workdir(server.server_id)
+        mv_dir = glob.glob(os.path.join(node_workdir, "data", "ks", "mv_cf_view-*"))[0]
+        delete_files_with_prefix(mv_dir)
+    # restart to avoid read cache
+    for server in servers:
+        await manager.server_restart(server.server_id)
```

Comment:
scylla should be stopped before you start deleting files

Reply:
Modified to stop first and then delete. But need to specify `durable_writes = false` when creating ks. If `durable_writes = true`, then the commitlog will be replayed at startup, causing the sstables of the deleted MV to be regenerated.